application/src/main/java/org/togetherjava/tjbot/commands/help/OnGuildLeaveCloseThreadListener.java

```diff
@@ -0,0 +1,72 @@
+package org.togetherjava.tjbot.commands.help;
+
+import net.dv8tion.jda.api.EmbedBuilder;
+import net.dv8tion.jda.api.entities.MessageEmbed;
+import net.dv8tion.jda.api.entities.ThreadChannel;
+import net.dv8tion.jda.api.events.GenericEvent;
+import net.dv8tion.jda.api.events.guild.member.GuildMemberRemoveEvent;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.togetherjava.tjbot.commands.EventReceiver;
+import org.togetherjava.tjbot.db.Database;
+import org.togetherjava.tjbot.db.generated.tables.AddHelpChannel;
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * Remove all thread channels associated to a user when they leave the guild
+ */
+public class OnGuildLeaveCloseThreadListener implements EventReceiver {
+    private static final Logger logger =
+            LoggerFactory.getLogger(OnGuildLeaveCloseThreadListener.class);
+    private final Database database;
+
+    public OnGuildLeaveCloseThreadListener(@NotNull Database database) {
+        this.database = database;
+    }
+
+    @Override
+    public void onEvent(@NotNull GenericEvent event) {
```

Comment:
extend ListenerAdapter and use onGuildMemberRemoveEvent(GuildMemberRemoveEvent) instead

Reply:
im not sure if this is possible with the current bot design. i vaguely remember having had problems with that

Reply:
Originally had the problem also so I stuck to this way. I can try to come up with a solution though. I have to look into what caused the original issue though. If I remember, it was not being able to add this to the features. Going to sleep now, but will update you tomorrow.

Reply:
> im not sure if this is possible with the current bot design. i vaguely remember having had problems with that

I don't see how it's an issue, the class doesn't extend anything else, it only implements the interface.

And ListenerAdapter implements EventListener, so don't think there will be an issue.


I only remember issues with SlashCommmands using the adapter, and you not being able to combine that win the ListenerAdapter

Reply:
Fixed it, thanks for the input. Much better way of getting the event